dotnet/src/Connectors/Connectors.Ollama/OllamaMetadata.cs

```diff
@@ -0,0 +1,145 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Runtime.CompilerServices;
+using OllamaSharp.Models;
+using OllamaSharp.Models.Chat;
+
+namespace Microsoft.SemanticKernel.Connectors.Ollama;
+
+/// <summary>
+/// Represents the metadata of the Ollama response.
+/// </summary>
+public sealed class OllamaMetadata : ReadOnlyDictionary<string, object?>
```

Comment:
This is new API surface that we don't expose in other connectors. We should discuss applying this pattern uniformly.

Reply:
This pattern was using in the `Google` connector, I may revert and drop this in Ollama one.